### ISSUE TITLE
memory consumption

### ISSUE THREAD

**Issue by kounkou** [kounkou/Data-Structures-and-Algorithms#6]
Take care of memory consumption.

**Comment by rajilsaj** [comment]
don't know how to implement that repo; what's the purpose ? :(

**Comment by kounkou** [comment]
Suppose you want to make a task manager. 
Suppose you want the task to be managed with the order of arrival.
Say : 
task 1, task 2, task 3
In that case, you could try to find a complex solution with anything attached to the task saying this task 1 is first, task 2 is second...
Or you could just implement a queue as I did.
The queue is implemented as circular buffer, you have 2 indexes (**begin**, and **end**), and those 2 give the place of the first task and end is consequently the last task.
Dequeuing those tasks is a simple Queue operation.
Hope this answers your question

**Comment by rajilsaj** [comment]
Too abstract for my taste

**Comment by kounkou** [comment]
Really ? 
I think that's because web devs don't tend to do much algorithms stuff. That might explain why it seems abstract to you. I think even Python don't usually need this...

**Comment by rajilsaj** [comment]
yes for sure, that's sure. Perhaps when my target in case of my future employment might change  I'd be more algo data structure guy like you.

**Comment by kounkou** [comment]
Data structure is bae ! 
I think it wouldn't be too wrong saying Data structures are to Computer science what Maths is to science.
The analogy is genuine.

**Comment by rajilsaj** [comment]
if it could help me to reach an opportunity abroad I'd be into as soon as possible

**Comment by kounkou** [comment]
AHAHA! sure it will help you. But you know, life ain't that simple anyway...

**Comment by rajilsaj** [comment]
ok you won I 'll get into... I found a book about Data analysis & Algo  with JS  (like it seems I'm diving in JS world this time) I 'll glance at that.

**Comment by kounkou** [comment]
Amazing man !
But now I am thinking about a project to use the data structures I learnt.
What about an artificial intelligence to digitize hand written text ?

**Comment by rajilsaj** [comment]
hummmm you 'd better be kiddding me

**Comment by kounkou** [comment]
Yeah man. It can be challenging but it's feasible. 
As an application, we could use that to automatically login on a WIFI network by scanning the code with the phone for instance.

**Comment by rajilsaj** [comment]
hummmmm ok create the repo I'd be waiting for ; it seem like you run away from skype those times. Job issues ?

**Comment by kounkou** [comment]
You're right, I have been working hard these days on features dev... You can ping me whenever you want. I am usually typing on my keyboard at the usual hours you know. I don't see you only though.
Okay, I will create the repo.
